cylc/flow/task_pool.py

```diff
@@ -705,12 +714,11 @@ def _get_task_by_id(self, id_):
             except KeyError:
                 pass
 
-    def queue_tasks(self, itasks):
-        """Queue tasks that are ready to run."""
-        for itask in itasks:
-            itask.state.reset(is_queued=True)
-            self.data_store_mgr.delta_task_queued(itask)
-        self.task_queue_mgr.push_tasks(itasks)
+    def queue_task(self, itask: TaskProxy) -> None:
+        """Queue a task that is ready to run."""
+        itask.state.reset(is_queued=True)
+        self.data_store_mgr.delta_task_queued(itask)
+        self.task_queue_mgr.push_task(itask)
```

Comment:
Might this be a good idea?
```suggestion
        if itask.state.reset(is_queued=True):
            self.data_store_mgr.delta_task_queued(itask)
            self.task_queue_mgr.push_task(itask)
```

Reply:
It certainly wouldn't hurt.